java/README.md

```diff
@@ -0,0 +1,101 @@
+# Java benchmarks
+
+## Verification Task Structure
+
+Verification tasks are grouped in directories, e.g.,
+`jbmc-regression`.  Within these directories, each verification task
+is in its own directory, e.g., `StringValueOf01_true-assert`. The
+suffix of this directory contains the expected verification outcome,
+i.e., `_true-assert` means that there is no execution of the program
+that violates any of the assertions in that program.  A verification
+task has a structure compatible with the [Maven](https://maven.apache.org)
+standard directory layout (to allow the future use of the Maven build
+system if necessary):
+
+| directory | description |
+| --------- | ----------- |
+|`src/` | |
+| `src/main/java/Main.java` | The `Main` class. |
+| `src/main/java/` | This directory contains the remaining Java source tree. |
+| `lib/` | This directory contains the library dependencies (without `java.*` and `javax.*`). |
+| `target/` | |
+| `target/classes/` | After compiling this directory contains the `.class` built from `src`. |
+| `target/task.jar` | This file assembles the `.class` files from `target/classes` and those from within the `.jar` files in `lib`. |
+
+The programs are assumed to be written in Java 1.8.
+The `.java` source files are in the source tree
+in the `src` sub-directory of the benchmark directory.  The
+program may call the Java standard library (`java.*`,
+`javax.*`).  Calls to other libraries are allowed under the
+condition that the library is provided as a `.jar` file in the
+`lib` sub-directory of the benchmark directory.  The benchmark
+must have a `Main` class with a `public static void
+  main(String[])` method in the root package. The `Main.java` file
+must have a copyright header indicating the source of the benchmark as
+well as name, version and links to the source code of all used libraries
+that are included in `lib`.
+
+## Compiling the Verification Tasks
+
+Makefiles for compiling the sources and assembling them into
+a `target/task.jar` file are provided through links to the
+`Makefile.task`.
+The top-level `Makefile` has a `compile` target for compiling all
+verification tasks. 
+
+## Preparing the Verification Tasks for Benchexec
+
+The input for a verification tool when run in Benchexec
+is a _verification-task-dir-name_`.zip` file, which contains
+the contents of a verification task directory, excluding compilation
+products. If a verification tool requires `.class` files or a `.jar`
+file as input it can use the `Makefile` provided to generate
```

Comment:
>  requiring the verifier to call `make

It doesn't require the verifier to call make. It says 'can use'.

Reply:
> Different Java compilers produce different byte code, and a verifier might require the source to be compiled with a specific compiler 

That's no problem as only the Java version is specfied and not a particular compiler.

Reply:
I know it says "can use `make`", but without the guarantees I mentioned above, it becomes a "must use `make`" in practice, because these guarantees are needed to be able to use the verification tasks without calling `make`.

Reply:
> "the verification task can be compiled by putting all files in the src directory on the sourcepath and all files from lib on the classpath"

Isn't this what the Makefile is saying? No problem. We can describe this also verbally. Anyway, this will simplify a bit as soon as I have removed the dependency jar files.

Reply:
Done.